tests/test_mlp_auto_sales.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023-2024 Retake, Inc.
```

Comment:
Can we name test theses a bit better? Perhaps something like `test_hive_partitioning` or something like that?